File: iosgl/iosgl/bt/material/DynamicBaseConstItem.m
//
//  DynamicBaseConstItem.m
//  iosgl
//
//  Created by zhao on 9/3/2020.
//  Copyright © 2020 zhao. All rights reserved.
//

#import "DynamicBaseConstItem.h"
#import "ConstItem.h"

@implementation DynamicBaseConstItem


-(void)setTargetInfo:(ConstItem*)target paramName:(NSString*)paramName type:(int)type;
{
    self.target = target;
    self.paramName = paramName;
    self.type = type;
    [self.target setDynamicOffset: self];
    self.currentValue=[[NSMutableArray alloc]init];
    for(int i=0;i<type;i++){
        [self.currentValue addObject:@0];
    }
    
}
-(void)setCurrentVal:(NSNumber*)x;
{
}
-(void)setCurrentVal:(NSNumber*)x y:(NSNumber*)y;
 {
 }
-(void)setCurrentVal:(NSNumber*)x y:(NSNumber*)y z:(NSNumber*)z;
{
}
@end

File: android/two/app/src/main/java/z3d/material/DynamicBaseConstItem.java
package z3d.material;

import java.util.ArrayList;
import java.util.List;

public class DynamicBaseConstItem {
    public ConstItem target;
    public String paramName ;
    public List<Float> currentValue;
    public int targetOffset;
    public int type;

    public void setTargetInfo(ConstItem $target,String $paramName,int $type) {
        this.target = $target;
        this.paramName = $paramName;
        this.type = $type;
        this.target.setDynamicOffset(this);
        this.currentValue = new ArrayList<>($type);

    }
    public void setCurrentVal(List args ) {
        for (int i = 0; i < args.size(); i++) {
            this.currentValue.add(i,(float)args.get(i));
        }
    }
}
